### PR TITLE
Updated utility.lua - Shortcuts for summon condition

### PR DESCRIPTION
**Pull request by EerieCode**
I've added to utility.lua a series of shortcuts for conditions like "If this card is Fusion Summoned..." or "target a Special Summoned monster...", similar to what we already use for Special Summon limits. Since many monsters have these conditions nowadays, this patch should facilitate scripting them.

### CODE REVIEW

**Comment by Cute-Nekomimi** [comment]
You can generalize those condition functions
`function Auxiliary.sum_con(t)
  return function(c)
    return bit.band(c:GetSummonType(),t)==t
  end
end`

Which would be used like aux.sum_con(TYPE_XYZ)

This can be further generalized to
`function Auxiliary.band_con(f,t,...)
  local args=table.pack(...)
  return function(c)
    return bit.band(f(c,table.unpack(args)),t)==t
  end
end`

Which would be used like, aux.band_con(Card.GetSummonType,SUMMON_TYPE_SYNCHRO)

You could alternate add
`function Card.IsSummonType(c,t)
return bit.band(c:GetSummonType(),t)==t
end`

Which would be used like
c:IsSummonType(SUMMON_TYPE_FUSION)

**Comment by EerieCode** [comment]
I thought about using something like the .sum_con() you suggested at first: I chose to keep it separate merely for consistency with the other functions I mentioned, the ones for the summon limit. If the mods for the repository find it too wordy, I'd have no issue with replacing it with that. I'm not entirely sure I understand how your second suggestion works, though: can you elaborate?

**Comment by Cute-Nekomimi** [comment]
If you mean band_con, it takes two required arguments, a function (lets call it "f") and a value. It then returns a function that takes a card as an argument, and when that function is called, it passes the card to f, and bit.bands the result of that function with the value you gave, and checks if it gives that value back. So you can do things like aux.band_con(Card.IsSummonType,TYPE_XYZ), or aux.band_con(Card.IsPreviousTypeOnField,TYPE_MONSTER), and those give you back functions that you can do in a check, for example Duel.SelectMatchingCard(aux.band_con(Card.IsSummonType,TYPE_XYZ),tp,0,LOCATION_MZONE,1,1,nil)

If you mean the third example, it basically adds an "IsSummonType" function to all cards, so you can use it like you would c:IsType or c:IsAttribute.

**Comment by EerieCode** [comment]
The last one would be the most useful, but that's a change to the core, not to utility.lua: since at the moment the only function that consistenly requires using bit.band is GetSummonType(), I think your first solution with sum_con() would be the most straightforward and simplest option.

**Comment by Cute-Nekomimi** [comment]
The last one (Card.IsSummonType) is actually doable in utility do lua, you just write
`function Card.IsSummonType(c,t)
return bit.band(c:GetSummonType,t)==t
end`
Into utility.lua
utility.lua just runs all of the code included in it, it doesn't have to directly work with the aux object, and you can add functions to Card, Group, and Effect just like you would define any other function
`function Card/Group/Effect.whatever`, and then you can call it with `thing:whatever`.

**Comment by EerieCode** [comment]
Thank you, I didn't know it worked that way! In this case, I'm closing this branch: I'll upload this function along with some of the new cards later, this way I can also test it with Starve Venom.
